integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,291 @@
+                    log.info("Found the Email with Subject : " + emailSubject);
+                    emailReceived = true;
+                    break;
+                }
+            }
+        } finally {
+            if (store != null) {
+                store.close();
+            }
+        }
+        return emailReceived;
+    }
+
+    /**
+     * Check inbox and make sure a particular email is deleted.
+     *
+     * @param emailSubject Email subject
+     * @param protocol     protocol used to connect to the server
+     * @return
+     * @throws MessagingException if we're unable to connect to the store
+     * @throws InterruptedException if thread sleep fails
+     */
+    public static boolean checkEmailDeleted(String emailSubject, String protocol, GreenMailUser user)
+            throws MessagingException, InterruptedException {
+        boolean isEmailDeleted = false;
+        long startTime = System.currentTimeMillis();
+
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (!isMailReceivedBySubject(emailSubject, EMAIL_INBOX, protocol, user)) {
+                log.info("Email has been deleted successfully!");
+                isEmailDeleted = true;
+                break;
+            }
+            Thread.sleep(500);
+        }
+        return isEmailDeleted;
+    }
+
+    /**
+     * Overloaded method to check deletion of mail in the primary user's store
+     * @param emailSubject
+     * @param protocol
+     * @return
+     * @throws MessagingException
+     * @throws InterruptedException
+     */
+    public static boolean checkEmailDeleted(String emailSubject, String protocol)
+            throws MessagingException, InterruptedException {
+        return checkEmailDeleted(emailSubject, protocol, primaryUser);
+    }
+
+    /**
+     * Check trash and make sure a particular email is moved to trash.
+     *
+     * @param emailSubject Email subject
+     * @param protocol     protocol used to connect to the server
+     * @return
+     * @throws MessagingException if we're unable to connect to the store
+     * @throws InterruptedException if thread sleep fails
+     */
+    public static boolean checkEmailMoved(String emailSubject, String protocol, GreenMailUser user)
+            throws MessagingException, InterruptedException {
+        boolean mailReceived = false;
+        long startTime = System.currentTimeMillis();
+        while ((System.currentTimeMillis() - startTime) < WAIT_TIME_MS) {
+            if (isMailReceivedBySubject(emailSubject, EMAIL_TRASH, protocol, user)) {
+                log.info("Found the moved email in mailbox : " + emailSubject);
+                mailReceived = true;
+                break;
+            }
+            Thread.sleep(500);
+        }
+        return mailReceived;
+    }
+
+    /**
+     * Overloaded method to check moving of mail in the primary user's store
+     * @param emailSubject
```

Comment:
Shall we add parameter descriptions? Let's correct in all places?

Reply:
done